Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -361,21 +361,30 @@ protected override DbCommand CreateDbCommand()
 
         protected override void Dispose(bool disposing)
         {
-            if (disposed)
-                return;
-
-            try
-            {
-                this.Close();
-            } 
-            catch (Exception ex)
+            if (!_disposed)
             {
-                // Prevent an exception from being thrown when disposing of this object
-                logger.Error("Unable to close connection", ex);
+                if (disposing)
+                {
+                    try
+                    {
+                        Close();
+                    }
+                    catch (Exception ex)
+                    {
+                        // Prevent an exception from being thrown when disposing of this object
+                        logger.Error("Unable to close connection", ex);
+                    }
+                }
+                else
+                {
+                    SfSession?.close();

```

Comment:
Don't we want exception handling here?

Reply:
I don't think this is necessary as the session Close method should be safe in the usage.